reconcile/terraform_repo.py

```diff
@@ -97,40 +100,57 @@ def run(
         )
 
         if repo_diff_result:
-            # put together output to pass to executor
-            actions_list: list[RepoOutput] = []
-
-            for repo in repo_diff_result:
-                actions_list.append(
-                    RepoOutput(
-                        repository=repo.repository,
-                        name=repo.name,
-                        ref=repo.ref,
-                        project_path=repo.project_path,
-                        delete=repo.delete or False,
-                        secret=RepoSecret(
-                            path=repo.account.automation_token.path,
-                            version=repo.account.automation_token.version,
-                        ),
-                    )
-                )
+            self.print_output(repo_diff_result, dry_run)
 
-            output = OutputFile(dry_run=dry_run, repos=actions_list)
+    def print_output(self, diff: list[TerraformRepoV1], dry_run: bool) -> None:
+        """Parses and prints the output of a Terraform Repo diff for the executor
 
-            if self.params.output_file:
-                try:
-                    with open(self.params.output_file, "w") as output_file:
-                        yaml.safe_dump(
-                            data=output.dict(),
-                            stream=output_file,
-                            explicit_start=True,
-                        )
-                except FileNotFoundError:
-                    raise ParameterError(
-                        f"Unable to write to '{self.params.output_file}'"
+        :param diff: list of terraform repos to be acted on
+        :type diff: list[TerraformRepoV1]
+        :param dry_run: whether the executor should perform a tf apply
+        :type dry_run: bool
+        """
+        actions_list: list[RepoOutput] = []
+
+        for repo in diff:
+            out_repo = RepoOutput(
+                repository=repo.repository,
+                name=repo.name,
+                ref=repo.ref,
+                project_path=repo.project_path,
+                delete=repo.delete or False,
+                secret=RepoSecret(
+                    path=repo.account.automation_token.path,
+                    version=repo.account.automation_token.version,
+                ),
+            )
+            # terraform-repo will store its statefiles in a specified directory if there is a
+            # terraform-state yaml file associated with the AWS account and a configuration is
+            # listed for terraform-repo, otherwise it will default to loading this information
+            # from the automation_token secret in Vault
```

Comment:
Under this default situation, the only difference is the statefile key will be different? Would the actual repo name still show up in the file name?

Reply:
Yes so each repo has its own TF statefile. In the default situation, each repo statefile would just be stored at the root directory of the S3 bucket rather than under a subdirectory